services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainMetrics.java

```diff
@@ -162,15 +174,62 @@ public BeaconChainMetrics(
             TekuMetricCategory.BEACON,
             "previous_correct_validators",
             "Number of validators who voted for correct source and target checkpoints in the previous epoch");
+
+    currentEpochParticipationWeight =
+        SettableGauge.create(
+            metricsSystem,
+            TekuMetricCategory.BEACON,
+            "current_epoch_participation_weight",
+            "Total effective balance of all validators who voted for correct source and target checkpoints in the current epoch");
+    previousEpochParticipationWeight =
+        SettableGauge.create(
+            metricsSystem,
+            TekuMetricCategory.BEACON,
+            "previous_epoch_participation_weight",
+            "Total effective balance of all validators who voted for correct source and target checkpoints in the previous epoch");
+
+    currentEpochTotalWeight =
+        SettableGauge.create(
+            metricsSystem,
+            TekuMetricCategory.BEACON,
+            "current_epoch_total_weight",
+            "Total effective balance of all active validators in the current epoch");
+    previousEpochTotalWeight =
+        SettableGauge.create(
+            metricsSystem,
+            TekuMetricCategory.BEACON,
+            "previous_epoch_total_weight",
+            "Total effective balance of all active validators in the previous epoch");
   }
 
   @Override
   public void onSlot(final UInt64 slot) {
-    recentChainData.getChainHead().ifPresent(this::updateMetrics);
+    recentChainData.getChainHead().ifPresent(head -> updateMetrics(slot, head));
   }
 
-  private void updateMetrics(final StateAndBlockSummary head) {
+  private void updateMetrics(final UInt64 slot, final StateAndBlockSummary head) {
     final BeaconState state = head.getState();
+    state
+        .getTransitionCaches()
+        .getLatestTotalBalances()
+        .ifPresent(
+            totalBalances -> {
+              // Update participation weights only once per epoch, a slot after epoch transition to
+              // make
+              // sure
+              // total balances are properly updated.
```

Comment:
Given these are now cheap to compute, we can probably update them each slot so that they update when we switch forks.

Reply:
Makes sense. Done.